gpii/node_modules/matchMakerFramework/src/MatchMakerUtilities.js

```diff
@@ -189,10 +189,10 @@ var fluid = fluid || require("infusion"),
 
             var leaves = gpii.matchMakerFramework.utils.computeLeaves(context.preferences);
             var disposed = strategy(leaves, tmpSolrecs, data, ontologyMetadata);
-            togo[contextId] = [];
+            togo[contextId] = {};
             fluid.each(disposed, function (solrec, solid) {
-                if (solrec.disposition === "accept") {
-                    togo[contextId].push(solid);
+                if (solrec.disposition === "accept" || solrec.disposition === "deactivate") {
```

Comment:
I don't think this model makes sense - a solution which needs to be deactivated should simply be "accept" together with a running state of false. "accept" means "we need to do something with this solution" and "reject" means "we don't need to do anything with this solution"